spec/providers/coreIntegration/tcpsocket.integration.src.js

```diff
@@ -70,10 +70,10 @@ module.exports = function (provider, setup) {
       }
       expect(evt).toEqual('onData');
       expect(msg.data.byteLength).toEqual(10);
-      done();
       socket.close(function () {});
       client.close(function () {});
       receiver.close(function () {});
+      done();
```

Comment:
Makes sense, I pushed something that I think addresses this but I'm not sure if I should try to pull in promises in some other way.